process/interceptors/factory/nilFinalityAttester.go

```diff
@@ -0,0 +1,31 @@
+package factory
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/logger"
+)
+
+var log = logger.GetOrCreate("process/interceptors/factory")
+
+// nilFinalityAttester is a nil implementation that will cause all received headers
+//not to be actually checked
+type nilFinalityAttester struct {
```

Comment:
I don't like the this mock in production code, this is only used in tests right?

I would move this to the mock folder instead
You also don't need any unit tests over a mock implementation

Reply:
Nil implementation removed, no longer required as we inject the implementation on the constructor